src/core/externalstorage/qgssimplecopyexternalstorage.cpp

```diff
@@ -0,0 +1,105 @@
+/***************************************************************************
+  qgssimplecopyexternalstorage.cpp
+  --------------------------------------
+  Date                 : March 2021
+  Copyright            : (C) 2021 by Julien Cabieces
+  Email                : julien dot cabieces at oslandia dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgssimplecopyexternalstorage_p.h"
+
+#include "qgscopyfiletask.h"
+#include "qgsapplication.h"
+
+#include <QFileInfo>
+
+QgsSimpleCopyExternalStorageStoredContent::QgsSimpleCopyExternalStorageStoredContent( const QString &filePath, const QString &url, const QString &authcfg )
+{
+  Q_UNUSED( authcfg );
+
+  mCopyTask = new QgsCopyFileTask( filePath, url );
```

Comment:
make_unique?

Reply:
The task is added to QgsTaskManager which owns it and will delete it, so we can't use a unique_ptr here (we would get a double delete)